src/toast/instrument_sim.py

```diff
@@ -459,22 +465,18 @@ def rhombus_layout(
     nullquat = np.array([0, 0, 0, 1], dtype=np.float64)
     rtthree = np.sqrt(3.0)
 
-    angwidth = angwidth * np.pi / 180.0
-    dim = rhomb_dim(npix)
+    dim = rhomb_dim(npos)
 
     # compute the height
-    angheight = rtthree * angwidth
+    angheight = rtthree * angwidth.to_value(u.radian)
 
     # find the angular packing size of one detector
-    pixdiam = angwidth / dim
-
-    # convert pol vector to radians
-    pol = polang * np.pi / 180.0
+    pixdiam = angwidth.to_value(u.radian) / dim
 
     # number of digits for pixel indexing
 
     ndigit = 0
-    test = npix
+    test = npos
     while test > 0:
```

Comment:
This could be `ndigit = int(np.log10(np.abs(test))) + 1`

Reply:
Implemented for both hex and rhombus layouts, with a slight variation.